src/inmanta/server/protocol.py

```diff
@@ -504,7 +504,7 @@ async def get_calls(self, no_hang: bool) -> Optional[List[common.Request]]:
             else:
                 timeout = self._interval
 
-            call = await self._queue.get(timeout=timedelta(seconds=timeout))
+            call = await self._queue.get(timeout=timedelta(seconds=timeout, microseconds=1))
```

Comment:
* We must add a clear comment regarding why adding one microsecond is required. I created an issue for this at Tornado, so you can also reference https://github.com/tornadoweb/tornado/issues/3271 in the comment. That way we know when we can remove the workaround.
* I would only add one microsecond when `timeout==0`. That way we keep the old behavior in production and only use the workaround when a specific test case requires it.